charts/redpanda/certs.go

```diff
@@ -89,7 +89,7 @@ func ClientCerts(dot *helmette.Dot) []certmanagerv1.Certificate {
 		panic(fmt.Sprintf("Certificate %q referenced but not defined", name))
 	}
 
-	if helmette.Empty(data.SecretRef) || !ClientAuthRequired(dot) {
+	if !ClientAuthRequired(dot) {
```

Comment:
I was a bit worried about this. We'll need to ask Azure (cc @c4milo or @ncole ) how they'd like to handle this. 
For mTLS to work, the chart needs to be able to get a client certificate for either console or RPK (@RafalKorepta which is it?). This currently works through the cert manager integration but there's not really a fallback. If y'all can mint a cert we can find a field to add a secretRef to or we can talk through relying on the cert-manger integration either partially or fully.

Reply:
I'm not sure the context on this, but I think you're saying for mTLS we need a client cert.  We are already creating a self-signed CA *AND* a client cert, so we can provide that ref somehow for Console (or RPK, whichever it is @RafalKorepta)?

In fact, in the other providers, if mTLS is enabled, we add a self-signed CA and a cert for kminion to use so we have an existing precedent for this.

Reply:
> For mTLS to work, the chart needs to be able to get a client certificate for either console or RPK (@RafalKorepta which is it?).

In this particular case, client certificate is used for both RPK and console. They will reference and mount the same certificate. 

> This currently works through the cert manager integration but there's not really a fallback. If y'all can mint a cert we can find a field to add a secretRef to or we can talk through relying on the cert-manger integration either partially or fully.

I agree, that this check needs to accommodate a use case when certificate is provided. I will change that check.

Reply:
> I agree, that this check needs to accommodate a use case when certificate is provided. I will change that check.

Let's handle that in a follow up PR/commit?